engine/utils/string_utils.h

```diff
@@ -29,4 +31,48 @@ inline std::vector<std::string> SplitBy(const std::string& str,
   } while (pos < str.length() && prev < str.length());
   return tokens;
 }
+
+inline uint64_t getCurrentTimeInMilliseconds() {
+  using namespace std::chrono;
+  return duration_cast<milliseconds>(system_clock::now().time_since_epoch())
+      .count();
+}
+
+inline std::string FormatTimeElapsed(uint64_t pastTimestamp) {
+  uint64_t currentTimestamp = getCurrentTimeInMilliseconds();
+  uint64_t milliseconds = currentTimestamp - pastTimestamp;
+
+  // Constants for time units
+  const uint64_t millisInSecond = 1000;
+  const uint64_t millisInMinute = millisInSecond * 60;
+  const uint64_t millisInHour = millisInMinute * 60;
+  const uint64_t millisInDay = millisInHour * 24;
+
+  uint64_t days = milliseconds / millisInDay;
+  milliseconds %= millisInDay;
+
+  uint64_t hours = milliseconds / millisInHour;
+  milliseconds %= millisInHour;
+
+  uint64_t minutes = milliseconds / millisInMinute;
+  milliseconds %= millisInMinute;
+
+  uint64_t seconds = milliseconds / millisInSecond;
+
+  std::ostringstream oss;
+
+  if (days > 0) {
```

Comment:
Data type is uint64_t, so this condition always satisfied?

Reply:
hmm, it can still == 0 right?

Reply:
ah, you're right, my bad.